api/README.md

```diff
@@ -8,12 +8,18 @@ Building
 
 Prerequisites:
 
-* JDK8+
-* Maven 3.0.3+
+* JDK 17+
+* Maven 3.2.5+
```

Comment:
As I said: https://github.com/jakartaee/concurrency/pull/477#discussion_r1572917378
```
[ERROR] Detected Maven Version: 3.2.5 is not in the allowed range [3.8.0,).
```

Reply:
Thanks :D Let me know if you see any other outdated versions.